src/main/java/org/opensearch/knn/quantization/models/quantizationParams/ScalarQuantizationParams.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationParams;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.knn.quantization.enums.ScalarQuantizationType;
+
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * The ScalarQuantizationParams class represents the parameters specific to scalar quantization (SQ).
+ * This class implements the QuantizationParams interface and includes the type of scalar quantization.
+ */
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor // No-argument constructor for deserialization
+@EqualsAndHashCode
+public class ScalarQuantizationParams implements QuantizationParams {
+    private ScalarQuantizationType sqType;
+    private static final long serialVersionUID = 1L; // Version ID for serialization
+
+    /**
+     * Static method to generate type identifier based on ScalarQuantizationType.
+     *
+     * @param sqType the scalar quantization type.
+     * @return A string representing the unique type identifier.
+     */
+    public static String generateTypeIdentifier(ScalarQuantizationType sqType) {
+        return generateIdentifier(sqType.getId());
+    }
+
+    /**
+     * Provides a unique type identifier for the ScalarQuantizationParams, combining the SQ type.
+     * This identifier is useful for distinguishing between different configurations of scalar quantization parameters.
+     *
+     * @return A string representing the unique type identifier.
+     */
+    @Override
+    public String getTypeIdentifier() {
+        return generateIdentifier(sqType.getId());
+    }
+
+    private static String generateIdentifier(int id) {
+        return String.format(Locale.ROOT, "ScalarQuantizationParams_%d", id);
```

Comment:
we use Locale.ROOT generally for exception messages as all users might not use English language. Given this identifier will be used as key, I think its better not to use locale and also not use String.format to keep identifiers consistent.

just do

```
return "ScalarQuantizationParams_" + id
```